Packs/SCCM/Integrations/SCCM/SCCM.ps1

```diff
@@ -0,0 +1,166 @@
+            $Creds = new-object -typename System.Management.Automation.PSCredential -argumentlist "$($Creds.DomainName)\$($Creds.UserName)", $SecurePassword
+            $progressPreference = 'silentlyContinue'
+            $result = Get-WmiObject -Class Win32_Product -ComputerName $deviceName -credential $Creds
+        }
+        $result
+    }
+    if ($Softwares)
+    {
+        $output = [PSCustomObject]@{
+            "SCCM.InstalledSoftwares" = $Softwares | ForEach-Object {[PSCustomObject]@{
+                Name = $_.Name
+                Version = $_.Version
+                Vendor = $_.Vendor
+                Caption = $_.Caption
+                IdentifyingNumber = $_.IdentifyingNumber.Trim('{}')
+            }
+        }
+        }
+        $MDOutput = $output."SCCM.InstalledSoftwares" | TableToMarkdown -Name "Installed softwares on $deviceName"
+        ReturnOutputs -ReadableOutput $MDOutput -Outputs $output -RawResponse $Softwares
+    }
+    else{
+        $output = @()
+        $MDOutput = $output | TableToMarkdown -Name "Installed softwares on $deviceName"
+        ReturnOutputs $MDOutput
+    }
+}
+
+
+Function TestModule($Session, $SiteCode, $Creds)
+{
+    Invoke-Command $Session -ArgumentList $SiteCode, $Creds -ErrorAction Stop -ScriptBlock {
+        param($SiteCode, $Creds)
+        Set-Location $env:SMS_ADMIN_UI_PATH\..\
+        Import-Module .\ConfigurationManager.psd1
+        Set-Location "$( $SiteCode ):"
+        if ((Get-Module -Name ConfigurationManager).Version -eq $null)
+        {
+            throw "Could not find SCCM modules in the SCCM machine"
+        }
+        $Devices = Get-CMResource -ResourceType System -Fast|Where-Object {$_.Name -ne $env:computername} | ForEach-Object {$_.Name}
+        # Checking Creds
+        if ($Devices){
+            $SecurePassword = ConvertTo-SecureString -AsPlainText -Force -String $Creds.Password
+            $Creds = new-object -typename System.Management.Automation.PSCredential -argumentlist "$($Creds.DomainName)\$($Creds.UserName)", $SecurePassword
+            $progressPreference = 'silentlyContinue'
+            $CheckCreds = Get-WmiObject -Class Win32_Product -ComputerName $Devices[0] -credential $Creds | Out-Null
+        }
+    }
+}
+
+function Main
+{
+    # Parse Params
+    $computerName = $demisto.Params()['ComputerName']
+    $key = $demisto.Params()['Key']
+    $userName = $demisto.Params()['UserName']
+    $port = $demisto.Params()['port']
+    $UseSSL = $demisto.Params()['insecure']
+    $SiteCode = $demisto.Params()['SiteCode']
+    $DomainName = $demisto.Params()['DomainName']
+    $password = $demisto.Params()['password']
+    $tmp = New-TemporaryFile
+    $key | Out-File $tmp.FullName
+    $Creds = @{DomainName=$DomainName;Password=$password;UserName=$userName}
+    $Session = New-PSSession -HostName $computerName -UserName $userName -Port $port -SSHTransport -KeyFilePath $tmp.FullName -ErrorAction Stop
```

Comment:
@glicht @hod-alpert why does by default PSremote uses the SSH Transport flag to create SSH Session to the remote endpoint and not via WinRM?
for this to work, OpenSSH and PS 6.0 core are required dependencies on the remote endpoints while WinRM is available on PS 5.1 which is OoB from Win 7 - Server 2019 and has its Secure session capabilities(https).
It is possible to use https://hub.docker.com/r/quickbreach/powershell-ntlm/ - Docker image of PowerShell with NTLM support to allow for PS-Remoting from Linux to Windows Over WinRM.

Reply:
@Tomer-M Is this doable?
If it is it will also solve the other problems we're experiencing

Reply:
it is doable, already tested to work with the quick breach docker image, does not have to be specifically this image tho, it just had to be linux distro with WSMAN support.
This also requires some level of configuration on the remote endpoints afaik from working with @ariansvirsky

Reply:
Have a look here:
https://blog.quickbreach.io/blog/powershell-remoting-from-linux-to-windows/
This is the guide we used, and we were able to establish session that way:
```
PS />    Enter-PSSession -ComputerName 82.166.99.236 -Authentication Negotiate -Credential $creds        
[82.166.99.236]: PS C:\Users\lab\Documents> .\PowerShell/
```
We were using user/pass combo but I'm sure there are another credentials you can use:
```
PS />  $creds = Get-Credential                                                                              
PowerShell credential request
Enter your credentials.
User: DESKTOP-S2455R8\lab
Password for user DESKTOP-S2455R8\lab: ****************
```

Reply:
works for me @hod-alpert 
![image](https://user-images.githubusercontent.com/60568193/93181172-ce570b00-f740-11ea-96dc-14dfbe202414.png)